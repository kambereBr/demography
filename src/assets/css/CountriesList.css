.filter-container > select {
  width: 100%;
}

.countries-container-ul {
  padding: 0;
  display: flex;
  flex-wrap: wrap;
  margin-top: 0;
}

.light {
  background-color: var(--bg-color) !important;
}

.dark {
  background-color: var(--primary-color) !important;
}

.filter-container select {
  height: 30px;
  border: 1px solid #999;
  font-size: 18px;
  color: #1c87c9;
  background-color: #eee;
  border-radius: 5px;
}
