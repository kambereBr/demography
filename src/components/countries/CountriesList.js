import { React, useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import Country from './Country';
import filterCountries from '../../utils/filters';
import '../../assets/css/CountriesList.css';
import FilterCountries from '../filter/FilterCountries';

const CountriesList = () => {
  const { countries } = useSelector((state) => state.countries);
  const [filteredCountries, setFilteredCountries] = useState([]);

  useEffect(() => {
    if (countries.length > 0) {
      setFilteredCountries(countries);
    }
  }, [countries]);

  const handleFilter = (value) => {
    setFilteredCountries(filterCountries(countries, value));
  };

  const colors = ['light', 'dark'];
  let ops = 'increment';
  let round = 0;
  const returnColor = () => {
    let color = '';
    if (ops === 'increment') {
      color = colors[round];
      round += 1;
      if (round > 1) {
        round = 1;
        ops = 'decrement';
      }
    } else {
      color = colors[round];
      round -= 1;
      if (round < 0) {
        round = 0;
        ops = 'increment';
      }
    }
    return color;
  };

  return (
    <div className="countries-container">
      <FilterCountries
        filterEvent={(e) => handleFilter(e.target.value)}
      />
      <div>
        <ul className="countries-container-ul">
          {filteredCountries.map((country) => (
            <Country
              key={country.iso3}
              country={country.country}
              iso3={country.iso3}
              code={country.code}
              populationCounts={country.populationCounts}
              classValue={returnColor() ?? ''}
            />
          ))}
        </ul>
      </div>
    </div>
  );
};

export default CountriesList;
