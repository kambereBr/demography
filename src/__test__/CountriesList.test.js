import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import CountriesList from '../components/countries/CountriesList';

describe('Render CountriesList correctly', () => {
  it('Render CountriesList Screen correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <CountriesList />
        </MemoryRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
