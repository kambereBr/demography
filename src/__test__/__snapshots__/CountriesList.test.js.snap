// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render CountriesList correctly Render CountriesList Screen correctly 1`] = `
<div
  className="countries-container"
>
  <div
    className="filter-container"
  >
    <select
      onChange={[Function]}
    >
      <option
        selected={true}
        value=""
      >
        Display All
      </option>
      <option
        value="A-E"
      >
        A to E
      </option>
      <option
        value="F-J"
      >
        F to J
      </option>
      <option
        value="K-O"
      >
        K to O
      </option>
      <option
        value="P-T"
      >
        P to T
      </option>
      <option
        value="U-Z"
      >
        U to Z
      </option>
    </select>
  </div>
  <div>
    <ul
      className="countries-container-ul"
    />
  </div>
</div>
`;
